libs/coin-framework/src/bot/specs.ts

```diff
@@ -78,13 +78,36 @@ export function deviceActionFlow<T extends TransactionCommon>(
           const stepValueTransform = currentStep.stepValueTransform || stepValueTransformDefault;
 
           if (!ignoreAssertionFailure && !disableStrictStepValueValidation) {
-            botTest("deviceAction confirm step '" + stepTitle + "'", () => {
-              expect({
-                [stepTitle]: stepValueTransform(stepValue),
-              }).toMatchObject({
-                [stepTitle]: expectedValue(arg, acc).trim(),
+            const expectedResult = expectedValue(arg, acc);
+
+            // Handle both sync and async expectedValue
+            if (expectedResult instanceof Promise) {
+              // For async expectedValue, perform validation asynchronously without blocking
+              const currentStepTitle = stepTitle;
+              const currentStepValue = stepValue;
+
+              expectedResult
+                .then(resolvedResult => {
+                  botTest("deviceAction confirm step '" + currentStepTitle + "'", () => {
+                    expect({
+                      [currentStepTitle]: stepValueTransform(currentStepValue),
+                    }).toMatchObject({
+                      [currentStepTitle]: resolvedResult.trim(),
+                    });
+                  });
+                })
+                .catch(error => {
+                  console.error(`Async validation failed for step '${currentStepTitle}':`, error);
+                });
+            } else {
+              botTest("deviceAction confirm step '" + stepTitle + "'", () => {
+                expect({
+                  [stepTitle]: stepValueTransform(stepValue),
+                }).toMatchObject({
+                  [stepTitle]: expectedResult.trim(),
+                });
```

Comment:
We could also just await as it works fine on sync functions

```suggestion
            const expectedResult = await expectedValue(arg, acc);

            botTest("deviceAction confirm step '" + stepTitle + "'", () => {
              expect({
                [stepTitle]: stepValueTransform(stepValue),
              }).toMatchObject({
                [stepTitle]: expectedResult.trim(),
              });
```